### ISSUE TITLE
Two instances of tmux using tmux-continuum on the same machine

### ISSUE THREAD

**Issue by lenneis** [tmux-plugins/tmux-continuum#16]
Hi,

thanks for resurrect/continuum, the are really needed functionality for tmux.  After reading https://github.com/tmux-plugins/tmux-continuum/blob/master/docs/multiple_tmux_servers.md, I think this might be the reason for this behaviour: If two different users on the same machine use tmux with resurrect/continuum enabled, only the first to start tmux gets the autosave feature enabled. So maybe you use some per machine locking mechanism. Is there a way to do this per user? On a multuser system this would be perfectly reasonable and safe as well.

**Comment by bruno-** [comment]
> thanks for resurrect/continuum, the are really needed functionality for tmux

Thanks, I'm glad you find them useful.

> Is there a way to do this per user?

I'm not sure, there might be a way. Frankly, I don't need this feature and I'm not the right candidate to implement it.

Let's see if other users would find this useful (please upvote with :+1:). If yea then we'd be looking for someone to implement this.

Just a note: when initially working on the feature of preventing 2 (or more) tmux servers doing autosave it was hard-ish to make it work properly. Adding process ownership to the mix could make this even more complicated.

**Comment by szimonek** [comment]
Hi,

We are also using one central access server for several users, where anyone can run tmux, so it was a problem also for us. This can be fixed with a simple and safe change. 
Edit .tmux/plugins/tmux-continuum/scripts/helpers.sh scroll to all_tmux_processes function and change it contents from:

```
all_tmux_processes() {
        # ignores `tmux source-file .tmux.conf` command used to reload tmux.conf
        ps -Ao "command pid" |
                \grep "^tmux" |
                \grep -v "^tmux source"
}
```


to:

```
all_tmux_processes() {
        # ignores `tmux source-file .tmux.conf` command used to reload tmux.conf
        ps -Ao "command user pid" |
                \grep `whoami` |
                \grep "^tmux" |
                \grep -v "^tmux source"
}
```

This function is used for searching if another tmux instance is running. By default it searches all processess and filters them only to tmux, so it catches all tmux instances started on machine. I have extended filter to match only tmux started by current user. This change is safe because each user has it's own save directory (default $HOME/.tmux/resurrect) and still only first tmux started by a user will save it's contents to this directory.

Feel free to create a pull request and merge this change to tmux-continuum code.

Globally allowing tmux-continuum to save multiple tmux instnaces is much more complicated because this will need multiple tmux-resurrect save directiories (one per tmux instnace), and some logic in tmux-continuum to choose right directory. I also agree with argument that normally running (and saving) multiple tmux instances is not required.

Best regards
Szymon

**Comment by bruno-** [comment]
Hi @szimonek thanks for the patch, I think it will do the work!
I applied it locally and will test it today. If all is good I'll push it by tomorrow.

**Comment by bruno-** [comment]
Implemented via 1531b3770a7cf7373d15fedd239c5331b99342d1

Sorry for the 3 year wait
